internal/automanagement/manager.go

```diff
@@ -27,7 +28,13 @@ func NewManager(instanceManager instancemgmt.InstanceManager) *Manager {
 func (m *Manager) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
 	h, err := m.Get(ctx, req.PluginContext)
 	if err != nil {
-		return nil, err
+		if len(req.Queries) == 0 {
+			// shouldn't be possible, but just in case
+			return nil, err
+		}
+		resp := backend.NewQueryDataResponse()
+		errorsource.AddErrorToResponse(req.Queries[0].RefID, resp, err)
```

Comment:
I think that the problem with this is that we are mixing real internal errors (like [failing to get a cache key](https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/instancemgmt/instance_manager.go#L94)) with [datasource errors](https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/instancemgmt/instance_manager.go#L133). We should probably differ between the different error types?

Also, if we are returning a valid DataResponse, I don't think we should also return the error, so here I would `return resp, nil`, as we do for the health check

Reply:
> I think that the problem with this is that we are mixing real internal errors (like [failing to get a cache key](https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/instancemgmt/instance_manager.go#L94)) with [datasource errors](https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/instancemgmt/instance_manager.go#L133).

Any errors that aren't explicitly marked with the "downstream" errorsource will be given a "plugin" error source, which distinguishes them in one way.

I could change this flow to only return an error response if the errorsource is already set. Would you prefer that, @andresmgot? Then any unhandled internal errors will return a `500` response and show only the opaque `An error occurred in the plugin` response to the user. @marefr would that also address your concern?

Reply:
> I could change this flow to only return an error response if the errorsource is already set. Would you prefer that, @andresmgot?

That sound good to me.

> @marefr would that also address your concern?

Not really, since the full datasource error is still returned to the user but again, we are already doing this for the CheckHealth scenario.